lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformDateTimeFormatterAndroid.java

```diff
@@ -204,8 +211,21 @@ else if (needTime)
   }
 
   @Override
-  public boolean isValidTimeZone(String timeZone) {
-    return TimeZone.getTimeZone(timeZone).getID().equals(timeZone);
```

Comment:
Previously, `normalizeTimeZoneName` would convert `"America/New_York"` to `"AMERICA/NEW_YORK"`
However, the problem is now `TimeZone.getTimeZone("AMERICA/NEW_YORK").getID()` => `"GMT"`